### ISSUE TITLE
Typings TS7016: Could not find a declaration file for module

### ISSUE THREAD

**Issue by Farnsi** [kazupon/vue-i18n#1670]
### Reporting a bug?

```ts
import { createI18n } from 'vue-i18n'
                            ^^^^^^^^
```

TS-Error

```
Could not find a declaration file for module 'vue-i18n'.
'/my-project/node_modules/vue-i18n/dist/vue-i18n.esm-bundler.js' implicitly has an 'any' type.
There are types at '/my-project/node_modules/vue-i18n/dist/vue-i18n.d.ts', but this result could
not be resolved when respecting package.json "exports".
The 'vue-i18n' library may need to update its package.json or typings.ts(7016)
```

### Expected behavior

Use .d.ts-file.

### Reproduction

https://jsfiddle.net/kazupon/rn724baz/3/

### System Info

```shell
vue-i18n: 9.2.2
vue: 3.2.47
typescript: 5.0.4
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/kazupon/vue-i18n/blob/v8.x/CONTRIBUTING.md)
- [X] Read the [Documentation](https://kazupon.github.io/vue-i18n/)
- [X] Check that there isn't [already an issue](https://github.com/kazupon/vue-i18n/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/kazupon/vue-i18n/discussions)
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by luisdeka** [comment]
Same error today

**Comment by Farnsi** [comment]
As soon as i remove the complete `"exports": {`block from package.json, it works.

No error and move over `import { createI18n } from 'vue-i18n'` or `$t('...` in VSCode shows the correct type definitions.

**Comment by EagleClark** [comment]
Same error 
vue-i18n: 9.2.2
vue: 3.2.47
typescript: 5.0.4

**Comment by luisdeka** [comment]
Fixed in beta version, as soon as they release new versions it will be fixed i suppose, at the time, change to v9.3

**Comment by larsrickert** [comment]
Hey, is it possible to release this as a quick fix for version 9.2? :)
The issue prevents us from using TypeScript 5

**Comment by koczko** [comment]
**Temporary solution.**
Work in my case, remove file: _tsconfig.node.json_

**Comment by kazupon** [comment]
This repository is vue-i18n v8, not v9
so close

**Comment by gabrielcaiana** [comment]
The problem was solved? I have the same problem in version `9.4.1`

nuxt `3.7.3`
node `18.16.0`

**Comment by larsrickert** [comment]
> The problem was solved? I have the same problem in version `9.4.1`
> 
> nuxt `3.7.3` node `18.16.0`

Its fixed for me with vue-i18n version >= 9.3.0 and TypeScript 5
